Content/Tiles/Dungeon/TwistSwordTile.cs

```diff
@@ -0,0 +1,126 @@
+//Sparkles
+//Glowmask
+
+using StarlightRiver.Content.Dusts;
+using StarlightRiver.Content.Items.Misc;
+using StarlightRiver.Core.Systems.DummyTileSystem;
+using System;
+using System.Linq;
+using Terraria.DataStructures;
+using Terraria.Enums;
+using Terraria.ID;
+using Terraria.ObjectData;
+
+namespace StarlightRiver.Content.Tiles.Dungeon
+{
+	public class TwistSwordTile : DummyTile
+	{
+		public override string Texture => AssetDirectory.Invisible;
+
+		public override int DummyType => ModContent.ProjectileType<TwistSwordTileProj>();
+		public override void SetStaticDefaults()
+		{
+			Main.tileFrameImportant[Type] = true;
+			Main.tileSolid[Type] = false;
+			TileObjectData.newTile.Height = 1;
+			TileObjectData.newTile.Width = 1;
+			TileObjectData.newTile.Origin = new Point16(0, 0); // Todo: make less annoying.
+			TileObjectData.newTile.AnchorTop = new AnchorData(AnchorType.SolidTile | AnchorType.SolidWithTop | AnchorType.SolidSide, TileObjectData.newTile.Width, 0);
+			TileObjectData.newTile.UsesCustomCanPlace = true;
+			TileObjectData.newTile.LavaDeath = false;
+			TileObjectData.newTile.CoordinateHeights = new int[] { 16 };
+			TileObjectData.newTile.CoordinateWidth = 16;
+			TileObjectData.newTile.CoordinatePadding = 2;
+			TileObjectData.addTile(Type);
+
+			HitSound = SoundID.Tink;
+
+			ModTranslation name = CreateMapEntryName();
+			name.SetDefault("Twisted Greatsword");
+			AddMapEntry(Color.Blue, name);
+		}
+
+		public override void KillTile(int i, int j, ref bool fail, ref bool effectOnly, ref bool noItem)
+		{
+			Item.NewItem(new EntitySource_TileBreak(i, j), i * 16, j * 16, 16, 16, ModContent.ItemType<TwistSword>());
+		}
+	}
+
+	public class TwistSwordTileProj : ModProjectile
+	{
+		public override string Texture => AssetDirectory.DungeonTile + "TwistSwordTile";
+
+		public float swaySpeed = 0.01f;
+
+		public float sway;
+
+		public float springiness = 0.1f;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Twisted Sword");
+		}
+
+		public override void SetDefaults()
+		{
+			Projectile.knockBack = 6f;
+			Projectile.width = 32;
+			Projectile.height = 72;
+			Projectile.penetrate = -1;
+			Projectile.ignoreWater = true;
+			Projectile.tileCollide = false;
+			Projectile.rotation = 1.57f;
+		}
+
+		public override bool PreDraw(ref Color lightColor)
+		{
+			Texture2D tex = ModContent.Request<Texture2D>(Texture).Value;
+			Texture2D glowTex = ModContent.Request<Texture2D>(Texture + "_Glow").Value;
+			Main.spriteBatch.Draw(tex, Projectile.position - Main.screenPosition, null, lightColor, Projectile.rotation - 1.57f, new Vector2(tex.Width / 2, 0), Projectile.scale, SpriteEffects.None, 0f);
+			Main.spriteBatch.Draw(glowTex, Projectile.position - Main.screenPosition, null, Color.White, Projectile.rotation - 1.57f, new Vector2(tex.Width / 2, 0), Projectile.scale, SpriteEffects.None, 0f);
+			return false;
+		}
+
+		public override void AI()
+		{
+			int i = (int)(Projectile.position.X / 16);
```

Comment:
Dummies have a ``Parent`` property for their parent tile, consider using that instead of manually finding it here